src/include/btree_inline.h

```diff
@@ -6,6 +6,73 @@
  * See the file LICENSE for redistribution information.
  */
 
+/*
+ * __wt_combined_addr_cookie_pack --
+ *     Pack a combined address cookie containing both block manager and page stat address cookie
+ *     parts.
+ */
+static inline int
+__wt_combined_addr_cookie_pack(WT_SESSION_IMPL *session, WT_ITEM *addr, void *block_addr,
+  uint8_t block_addr_size, WT_PAGE_STAT ps)
+{
+    /* Initialize the combined address cookie buffer. */
+    WT_RET(__wt_buf_init(session, addr, WT_ADDR_COOKIE_MAX));
+
+    WT_ADDR_COOKIE_BLOCK_LEN(addr->mem) = block_addr_size;
+    memcpy(WT_ADDR_COOKIE_BLOCK(addr->mem), block_addr, block_addr_size);
+    WT_RET(__wt_addr_cookie_page_stat_pack(addr->mem, ps));
+
+    /* Update the size of the combined address cookie. */
+    addr->size = (uint8_t)(
+      1 + WT_ADDR_COOKIE_BLOCK_LEN(addr->mem) + 1 + WT_ADDR_COOKIE_PAGE_STAT_LEN(addr->mem));
+    return (0);
+}
+
+/*
+ * __wt_combined_addr_cookie_unpack --
+ *     Unpack a combined address cookie containing both block manager and page stat address cookie
+ *     parts.
+ */
+static inline int
+__wt_combined_addr_cookie_unpack(const void *addr, WT_CELL_UNPACK_COMMON *unpack, WT_PAGE_STAT *ps)
```

Comment:
I don't think we should unpack the whole thing together. One of the benefit of putting data in the address cookie is that we only need to unpack the data we need and skip other data.

Reply:
Understood. I've removed the `__wt_combined_addr_cookie_unpack` function here: [8754ca1](https://github.com/wiredtiger/wiredtiger/pull/8792/commits/8754ca1e57dd67cd35eaf554bac288d8e7a23d67). We can simply use the `__wt_addr_cookie_page_stat_unpack` function to unpack only the page stats when we get around to implementing the aggregation logic.